app/models/loader/types.rb

```diff
@@ -148,10 +148,23 @@ def create!
       protected
 
       def layer_roleids
-        self.layers.map do |layer|
+        verify_layers_exist!
+
+        Array(self.layers).map do |layer|
```

Comment:
We need `Array` here because this can either be a single Layer or and array containing multiple Layers.